Architecture.md

```diff
@@ -0,0 +1,193 @@
+<!-- This file contains a rough overview of the PyO3 codebase. -->
+<!-- Please do not make descriptions too specific, so that we can easily -->
+<!-- keep this file in sync with the codebase. -->
+
+# PyO3: Architecture
+
+This document roughly describes the high-level architecture of PyO3.
+If you want to become familiar with the codebase you are in the right place!
+
+## Overview
+
+PyO3 provides a bridge between Rust and Python, based on the [Python C/API].
```

Comment:
`[Python C/API]` this link seems broken: https://github.com/PyO3/pyo3/blob/bbca585002a153125ab586725b9adcbb4bd8c273/Architecture.md

Reply:
BTW, I think it should be "Python/C API" not "Python C/API".

Reply:
Thanks!